services/rds/src/main/java/software/amazon/awssdk/services/rds/DefaultRdsUtilities.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.rds;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.signer.Aws4Signer;
+import software.amazon.awssdk.auth.signer.params.Aws4PresignerParams;
+import software.amazon.awssdk.awscore.client.config.AwsClientOption;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.rds.model.GenerateAuthenticationTokenRequest;
+import software.amazon.awssdk.utils.StringUtils;
+
+@Immutable
+@SdkInternalApi
+final class DefaultRdsUtilities implements RdsUtilities {
+    private final Region region;
+    private final AwsCredentialsProvider credentialsProvider;
+
+    DefaultRdsUtilities(DefaultBuilder builder) {
+        this.credentialsProvider = builder.credentialsProvider;
+        this.region = builder.region;
+    }
+
+    /**
+     * Used by RDS low-level client's utilities() method
+     */
+    @SdkInternalApi
+    static RdsUtilities create(SdkClientConfiguration clientConfiguration) {
+        return new DefaultBuilder().clientConfiguration(clientConfiguration).build();
+    }
+
+    @Override
+    public String generateAuthenticationToken(GenerateAuthenticationTokenRequest request) {
+        SdkHttpFullRequest httpRequest = SdkHttpFullRequest.builder()
+                                            .method(SdkHttpMethod.GET)
+                                            .protocol("https")
+                                            .host(request.hostname())
+                                            .port(request.port())
+                                            .encodedPath("/")
+                                            .putRawQueryParameter("DBUser", request.userName())
+                                            .putRawQueryParameter("Action", "connect")
+                                            .build();
+
+        Instant expirationTime = Instant.now().plus(15, ChronoUnit.MINUTES);
```

Comment:
nit, can we make it a constant and add a comment with the [link](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html ) there so that it's less of a magic number?

Reply:
Yep